pkg/engine/armresources.go

```diff
@@ -0,0 +1,165 @@
+
+	//for _, profile := range cs.Properties.AgentPoolProfiles {
+	//	vmss := CreateAgentVMSS(cs, profile)
+	//	armResources = append(armResources, vmss)
+	//}
+	//
+	//masterAVProfile := cs.Properties.MasterProfile.AvailabilityProfile
+	//
+	//if masterAVProfile == api.AvailabilitySet {
+	//	masterAvSet := CreateAvailabilitySet(cs)
+	//	armResources = append(armResources, masterAvSet)
+	//	masterVM := CreateVirtualMachine(cs)
+	//	armResources = append(armResources, masterVM)
+	//} else {
+	//	masterVmss := CreateMasterVMSS(cs)
+	//	armResources = append(armResources, masterVmss)
+	//}
+	//
+	//publicIp := CreatePublicIPAddress(cs)
+	//armResources = append(armResources, publicIp)
+	//loadBalancers := CreateLoadBalancer()
+	//masterNIC := CreateNetworkInterfaces(cs)
+	//armResources = append(armResources, loadBalancers)
+	//armResources = append(armResources, masterNIC)
+	//
+	//if cs.GetCloudSpecConfig().CloudName == api.AzurePublicCloud {
+	//	aksBillingExtension := CreateAKSBillingExtension(cs)
+	//	armResources = append(armResources, aksBillingExtension)
+	//}
+
+	return armResources
+}
+
+func createKubernetesAgentVMASResources(cs *api.ContainerService, profile *api.AgentPoolProfile) []interface{} {
+	var agentVMASResources []interface{}
+
+	if profile.IsWindows() {
+		if cs.Properties.WindowsProfile.HasCustomImage() {
+			agentVMASResources = append(agentVMASResources, createWindowsImage(cs, profile))
+		}
+	}
+
+	agentVmasNic := createAgentVMASNIC(cs, profile)
+	agentVMASResources = append(agentVMASResources, agentVmasNic)
+
+	if profile.IsManagedDisks() {
+		agentAvSet := createAgentAvailabilitySets(cs, profile)
+		agentVMASResources = append(agentVMASResources, agentAvSet)
+	} else if profile.IsStorageAccount() {
+		agentStorageAccount := createAgentVMASStorageAccount(cs, profile, false)
+		agentVMASResources = append(agentVMASResources, agentStorageAccount)
+		if profile.HasDisks() {
+			agentDataDiskStorageAccount := createAgentVMASStorageAccount(cs, profile, true)
+			agentVMASResources = append(agentVMASResources, agentDataDiskStorageAccount)
+		}
+
+		avSet := AvailabilitySetARM{
+			ARMResource: ARMResource{
+				APIVersion: "[variables('apiVersionCompute')]",
+			},
+			AvailabilitySet: compute.AvailabilitySet{
+				Location: to.StringPtr("[variables('location')]"),
+				Name: to.StringPtr(fmt.Sprintf("[variables('%sAvailabilitySet')]",
+					profile.Name)),
+				AvailabilitySetProperties: &compute.AvailabilitySetProperties{},
+				Type:                      to.StringPtr("Microsoft.Compute/availabilitySets"),
+			},
+		}
+
+		agentVMASResources = append(agentVMASResources, avSet)
+	}
+
+	agentVmasVM := createAgentAvailabilitySetVM(cs, profile)
+	agentVMASResources = append(agentVMASResources, agentVmasVM)
+
+	useManagedIdentity := cs.Properties.OrchestratorProfile.KubernetesConfig.UseManagedIdentity
+	userAssignedIDEnabled := useManagedIdentity && cs.Properties.OrchestratorProfile.KubernetesConfig.UserAssignedID != ""
```

Comment:
Seen this multiple places : Can we have a func on KubernetesConfig instead?
`cs.Properties.OrchestratorProfile.KubernetesConfig.IsUserAssignedIdentityEnabled()`

Reply:
Great Idea! I'll push this to aks-engine upstream